src/librbd/image/CreateRequest.cc

```diff
@@ -301,6 +307,69 @@ void CreateRequest<I>::handle_validate_data_pool(int r) {
     return;
   }
 
+  validate_group();
```

Comment:
I don't think  image creation should be tied to a group. That should be a 2 step operation - 1) create  image
2) add image to group (probably in the Group class).

Reply:
This is happening as part of `validate_data_pool` and we only set group image options on the image while the initial creation. There will be window in slipping this op, which might lead to unnecessary races.

I think it should be OK, unless you foresee any potential issues.

Reply:
I don't remember already, but I believe the main reason for adding this ability was for the rbd-mirror to bootstrap  the secondary group, where it would be very problematic (if possible at all) to do this properly in two steps. And as a bonus we have an ability to do this for the primary group, which I think is a good thing. I don't see why doing this in two steps is preferable.

Reply:
IMO, we should not tie the image creation to a group as there is ideally no connection between them at the time if creation.